packages/flutter/test/rendering/slivers_helpers_test.dart

```diff
@@ -0,0 +1,84 @@
+      expect(applyGrowthDirectionToAxisDirection(axisDirection, GrowthDirection.forward), axisDirection);
+    expect(applyGrowthDirectionToAxisDirection(AxisDirection.up, GrowthDirection.reverse), AxisDirection.down);
+    expect(applyGrowthDirectionToAxisDirection(AxisDirection.down, GrowthDirection.reverse), AxisDirection.up);
+    expect(applyGrowthDirectionToAxisDirection(AxisDirection.left, GrowthDirection.reverse), AxisDirection.right);
+    expect(applyGrowthDirectionToAxisDirection(AxisDirection.right, GrowthDirection.reverse), AxisDirection.left);
+  });
+
+  test('SliverConstraints', () {
+    SliverConstraints a = new SliverConstraints(
+      axisDirection: AxisDirection.down,
+      growthDirection: GrowthDirection.forward,
+      userScrollDirection: ScrollDirection.idle,
+      scrollOffset: 0.0,
+      overlap: 0.0,
+      remainingPaintExtent: 0.0,
+      crossAxisExtent: 0.0,
+    );
+    SliverConstraints b = a.copyWith();
+    expect(a, equals(b));
+    expect(a.hashCode, equals(b.hashCode));
+    expect(a.toString(), equals(b.toString()));
+    expect(a.scrollOffsetAsOffset, equals(const Offset(0.0, 0.0)));
+    expect(a.normalizedGrowthDirection, equals(GrowthDirection.forward));
+
+    SliverConstraints c = a.copyWith(
+      axisDirection: AxisDirection.up,
+      growthDirection: GrowthDirection.reverse,
+      userScrollDirection: ScrollDirection.forward,
+      scrollOffset: 10.0,
+      overlap: 20.0,
+      remainingPaintExtent: 30.0,
+      crossAxisExtent: 40.0,
+    );
+    SliverConstraints d = new SliverConstraints(
+      axisDirection: AxisDirection.up,
+      growthDirection: GrowthDirection.reverse,
+      userScrollDirection: ScrollDirection.forward,
+      scrollOffset: 10.0,
+      overlap: 20.0,
+      remainingPaintExtent: 30.0,
+      crossAxisExtent: 40.0,
+    );
+    expect(c, equals(d));
+    expect(c.scrollOffsetAsOffset, equals(const Offset(0.0, -10.0)));
+    expect(c.normalizedGrowthDirection, equals(GrowthDirection.forward));
+    expect(d.scrollOffsetAsOffset, equals(const Offset(0.0, -10.0)));
+    expect(d.normalizedGrowthDirection, equals(GrowthDirection.forward));
+
+    SliverConstraints e = d.copyWith(axisDirection: AxisDirection.right);
+    expect(e.scrollOffsetAsOffset, equals(const Offset(10.0, 0.0)));
+    expect(e.normalizedGrowthDirection, equals(GrowthDirection.reverse));
+
+    SliverConstraints f = d.copyWith(axisDirection: AxisDirection.left);
+    expect(f.scrollOffsetAsOffset, equals(const Offset(-10.0, 0.0)));
+    expect(f.normalizedGrowthDirection, equals(GrowthDirection.forward));
+
+    SliverConstraints g = d.copyWith(growthDirection: GrowthDirection.forward);
+    expect(g.scrollOffsetAsOffset, equals(const Offset(0.0, -10.0)));
+    expect(g.normalizedGrowthDirection, equals(GrowthDirection.reverse));
+  });
+
+  test('SliverGeometry', () {
+    expect(() {
+      new SliverGeometry().debugAssertIsValid;
+    }, isNot(throws));
```

Comment:
Why not just:

`expect(new SliverGeometry().debugAssertIsValid(), isTrue)`

?  The test will fail if that throws.